### ISSUE TITLE
Allow users to set event endpoint manually

### ISSUE THREAD

**Issue by mcnaveen** [umami-software/umami#2347]
### Describe the feature or enhancement

TLDR: Basically I want to hide the umami url under my main domain. Which is working fine. But the data is not properly sent to `/api/send`

Umami makes it difficult to proxy via Nginx.

Plausible has a script tag like this.
```
<script defer data-api="/api/event" data-domain="website.com" src="/js/script.js"></script>
```

If you look at this they are letting us set `data-api` which is an event endpoint.

But in Umami case, the event endpoint is `/api/send` where the root domain is extracted from the script `src` URL by removing the script.js part

The problem is. Even if we proxy it like this

```
    set $umami_script_url https://umami.domain.com/script.js;
    location = /js/script.js {
        proxy_pass $umami_script_url;
        proxy_set_header Host umami.domain.com;
        proxy_buffering on;
        add_header X-Cache $upstream_cache_status;
    }
```
data for the `/api/send` is set to the current client domain name only. Not to the umami instance URL.

Ref: https://plausible.io/docs/proxy/guides/nginx
